client/device/asa/update_test.go

```diff
@@ -312,7 +313,11 @@ func TestAsaUpdate(t *testing.T) {
 
 			testCase.setupFunc(testCase.input)
 
-			output, err := asa.Update(context.Background(), *http.MustNewWithDefault("https://unittest.cdo.cisco.com", "a_valid_token"), testCase.input)
+			output, err := asa.Update(
+				context.Background(),
+				*http.MustNewWithConfig(baseUrl, "a_valid_token", 0, 0, time.Minute),
```

Comment:
I see the config is the same across all tests, perhaps we could consider creating a generic helper for tests that encapsulates all the common elements

Reply:
Yes that would be a good refactor if it becomes messy. Let's do it in another PR since it is already quite big here